### PR TITLE
Time to go, gumballs and nano potions

### PR DESCRIPTION
**Pull request by kamilprzyb**
- "Go to the time machine" button is functional
- Warping to the past sets PayZoneFlag
- and quits all active missions
- Implemented gumballs
- Implemented nano potions

### CODE REVIEW

**Comment by Ssbhawk** [comment]
What do you mean by "PayZoneFlag"?

**Comment by Raymonf** [comment]
@Ssbhawk `PCStyle2.iPayzoneFlag`

File: src/MissionManager.cpp [hunk 1]
```diff
@@ -157,13 +157,17 @@ void MissionManager::quitMission(CNSocket* sock, CNPacketData* data) {
         return; // malformed packet
 
     sP_CL2FE_REQ_PC_TASK_STOP* missionData = (sP_CL2FE_REQ_PC_TASK_STOP*)data->buf;
+    quitTask(sock, missionData->iTaskNum);
+}
+
+void MissionManager::quitTask(CNSocket* sock, int32_t TaskNum) {
```
Comment (Raymonf) [inline src/MissionManager.cpp:163]: can we make this camel-cased? `taskNum`

File: src/PlayerManager.hpp [hunk 2]
```diff
@@ -50,7 +50,7 @@ namespace PlayerManager {
     void setSpecialPlayer(CNSocket* sock, CNPacketData* data);
     void heartbeatPlayer(CNSocket* sock, CNPacketData* data);
     void revivePlayer(CNSocket* sock, CNPacketData* data);
-    void exitGame(CNSocket* sock, CNPacketData* data);
+    void exitGame(CNSocket* sock, CNPacketData* data);
```
Comment (Raymonf) [inline src/PlayerManager.hpp:53]: 4 spaces at the end

File: src/MissionManager.cpp (hunk 1 above)
Comment (kamilprzyb) [reply src/MissionManager.cpp]: fixed

File: src/PlayerManager.hpp (hunk 2 above)
Comment (kamilprzyb) [reply src/PlayerManager.hpp]: fixed

**Comment by Ssbhawk** [comment]
I meant like "What IS a PayZoneFlag"? Does it involve me paying actual money?

**Comment by CPunch** [comment]
> I meant like "What IS a PayZoneFlag"? Does it involve me paying actual money?

No, the original game was locked behind a payzone flag. It's just used to tell whether a character should be in the (free) Future map, or the (paid) Past map of the original game.

**Comment by Ssbhawk** [comment]
Ah! Got it!

**Pull request merged**